src/components/pagination/CdrPagination.vue

```diff
@@ -1,3 +1,170 @@
+
+<script setup>
+import { useCssModule, defineEmits, computed, ref, watch, nextTick, onMounted } from 'vue';
+import mapClasses from '../../utils/mapClasses';
+import propValidator from '../../utils/propValidator';
+import IconCaretLeft from '../icon/comps/caret-left';
+import IconCaretRight from '../icon/comps/caret-right';
+import CdrSelect from '../select/CdrSelect';
+import uid from '../../utils/uid';
+
+const props = defineProps({
+    // NOTE pagination now requires an ID
+    id: {
+      type: String,
+    },
+    /**
+     * Total number of pages. Sometimes the total number of pages is different than total page data
+     * and this prop allows for that.
+     */
+    totalPages: {
+      type: Number,
+      default: null,
+    },
+    /**
+     * Array of objects. Objects have structure of { page: Number, url: String }
+     */
+    pages: {
+      type: Array,
+      required: true,
+      validator: (value) => {
+        const result = value.every((obj) => {
+          if (!Object.prototype.hasOwnProperty.call(obj, 'page')
+          || typeof obj.page !== 'number') {
+            console.error('Property "page" is missing or is not a number', obj); // eslint-disable-line
+            return false;
+          } if (!Object.prototype.hasOwnProperty.call(obj, 'url')
+          || typeof obj.url !== 'string') {
+            console.error('Property "url" is missing or is not a string', obj); // eslint-disable-line
+            return false;
+          }
+          return true;
+        });
+        return result;
+      },
+    },
+    linkTag: {
+      type: String,
+      default: 'a',
+      validator: (value) => propValidator(
+        value,
+        ['a', 'button'],
+      ),
+    },
+    forLabel: {
+      type: String,
+      default: '',
+    },
+    /** @ignore used for binding v-model, represents the current page */
+    modelValue: {
+      type: Number,
+    },
+  })
+  const emit = defineEmits(['update:modelValue', 'navigate']);
+  const uniqueId = props.id ? props.id : uid();
+  const currentIdx = ref(0);
+  const linkRefs = ref([]);
+  const innerValue = computed({
+    get: () => props.modelValue,
+    set: val => {
+      setCurrentIdx(val);
+      emit('update:modelValue', val);
+    }
+  });
+
+  const setCurrentIdx = (page) => {
+    currentIdx.value = props.pages.map((x) => x.page).indexOf(page);
+  }
+
+  const navigate = (pageNum, e) => {
+    // TODO: provide option to just emit without navigating to support vue-router?
+    // Dont do anything if clicking the current active page
+    const paginationLinkIsVisible = e.target.offsetWidth > 0 && e.target.offsetHeight > 0;
```

Comment:
When the pagination link is visible, we should expect the default behavior when someone clicks the pagination link for the current page.

The pagination link will not be visible, however, when we're using the select element instead of a list of links (this is in mobile view widths). In this case a click is simulated on the hidden link and we need the navigate event to emit.